### PR TITLE
Added ability to generate metadata from EarlyBound entities

### PR DESCRIPTION
**Pull request by BetimBeja**
Hello @jordimontana82 ,
this PR implements the functionality requested in issue #247.
One thing it is missing is mapping the Primary Id Attribute to UniqueIdentifierAttributeMetadata. Actually it produces EntityReference, since it gets mapped from a Nullable<Guid> type.
I believe that with this implementation we can start to use EntityMetadata in our algorithms for both EarlyBound and LateBound entities and probably we can even mix them without too much problems.
I used some code from @eNeRGy164 implementation but I don't know how to give him credit other than mentioning him 😄 
Best regards,
Betim.

### CODE REVIEW

**Comment by jordimontana82** [comment]
Great @BetimBeja  thx a mill for this.

File: FakeXrmEasy.Shared/Metadata/MetadataGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        private static T GetCustomAttribute<T>(MemberInfo member) where T : Attribute
+        {
+            return (T)Attribute.GetCustomAttribute(member, typeof(T));
+        }
+
+        private static AttributeMetadata CreateAttributeMetadata(Type propertyType)
+        {
+            switch (propertyType.Name)
+            {
+                case "String":
```
Comment (jordimontana82) [inline FakeXrmEasy.Shared/Metadata/MetadataGenerator.cs]: Just some giving some feedback... and this is totally subjective but I'd rather prefer comparing using the property type as opposed to use magic strings? i.e. 

    if (attributeType == typeof(string)) {

     }

I also believe it would be nice to add one unit test per data type so we don't miss any edge case (i.e. String type in early bound to String Metadata, OptionSetValue to PicklistAttributeMetadata etc...

Thoughts?

Comment (BetimBeja) [reply FakeXrmEasy.Shared/Metadata/MetadataGenerator.cs]: Hello Jordi,
I refactored the function to remove the magic strings, and I introduced some other magic strings in the process 😄 for Id, statecode and statuscode.
I will add some more unit tests for each type... my Surface goes on fire every time I open the GeneratedCode.cs file 😈 so it will take some time to find one attribute per data type I guess.
